org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java

```diff
@@ -80,6 +88,37 @@ public SignatureHelp signatureHelp(TextDocumentPositionParams position, IProgres
 		return help;
 	}
 
+	private boolean isValid(ICompilationUnit unit, int offset, IProgressMonitor monitor) {
+		CompilationUnit ast = CoreASTProvider.getInstance().getAST(unit, CoreASTProvider.WAIT_YES, monitor);

```

Comment:
instead of traversing the whole document, couldn't we find the closest enclosing node instead?

Reply:
I'm not sure how to do that. Do you have any idea?